pyodide_build/pywasmcross.py

```diff
@@ -233,42 +261,22 @@ def replay_genargs_handle_argument(arg: str) -> str | None:
     if arg.startswith("-L/usr"):
         return None
 
-    # fmt: off
-    if arg in [
-        # threading is disabled for now
-        "-pthread",
-        # this only applies to compiling fortran code, but we already f2c'd
-        "-ffixed-form",
-        "-fallow-argument-mismatch",
-        # On Mac, we need to omit some darwin-specific arguments
-        "-bundle", "-undefined", "dynamic_lookup",
-        # This flag is needed to build numpy with SIMD optimization which we currently disable
-        "-mpopcnt",
-        # gcc flag that clang does not support
-        "-Bsymbolic-functions",
-        '-fno-second-underscore',
-        '-fstack-protector',  # doesn't work?
-        '-fno-strict-overflow',  # warning: argument unused during compilation
-        "-mno-sse2", # warning: argument unused during compilation
-        "-mno-avx2", # warning: argument unused during compilation
-        "-std=legacy", # fortran flag that clang does not support
-    ]:
+    if arg in EXCLUDED_ARGUMENTS:
         return None
 
-    if arg.startswith((
-        "-J",  # fortran flag that clang does not support
-    )):
+    if arg.startswith(EXCLUDED_ARG_PREFIXES):
         return None
 
-    # fmt: on
     return arg
 
 
+@functools.cache
```

Comment:
As I understand, the CMake compiler flags are calculated just once as this is a fixed mapping, the symlinks directory isn't changing, and they won't change between different invocations either.